### DISCUSSION TITLE
v0.5.0-beta.1

### DISCUSSION THREAD

**Discussion by mgdigital** [bitmagnet-io/bitmagnet#109] [Announcements]
**Thanks for giving this beta a try and reporting any issues!**

v0.5.0 focuses on text search performance. The way torrents are indexed and queried for text search has been completely reworked; the final release will include some documentation updates for the website that will go into further detail.

## Upgrading

The v0.5.0 upgrade process is completely automatic (unless you've used the `/import` endpoint, read below). All torrents must be reprocessed before they will work with text search. Just wait for the background job queue to work through them, then enjoy better and faster search! 

The queue may take several hours to work down if you have a large library. To check on progress, visit the `/metrics` endpoint and note metrics that begin `asynq_tasks_enqueued_total{queue="process_torrent"`.

**Note:** I'm aware there are still some performance issues that need addressing for aggregations. These will be addressed separately - this release focuses on text search.

**If you've used the `/import` endpoint**, you will lose any classification hints that were part of the import. That is, if torrents were assigned to particular content types, or with an IMDB ID, this information will be lost and the classifier will attempt to classify from scratch without these hints. To recover the hints, the original import data must be reimported - it was necessary to separate these hints from the final classification result as the 2 things were previously conflated, making it impossible to separate out the data as part of this upgrade. **IMPORTANT**: the format for the `/import` endpoint has changed slightly, and the documentation for this on a branch here during the beta phase: https://github.com/bitmagnet-io/bitmagnet/blob/0.5.0-docs/bitmagnet.io/tutorials/import.md (note there is a slight change to the `jq` query in the final code snippet).

## What's Changed
* Bump golang.org/x/crypto from 0.14.0 to 0.17.0 by @dependabot in https://github.com/bitmagnet-io/bitmagnet/pull/91
* Lazy instantiation for services with side effects by @mgdigital in https://github.com/bitmagnet-io/bitmagnet/pull/98
* Search query string rework by @mgdigital in https://github.com/bitmagnet-io/bitmagnet/pull/96
* Cache tweaks (increase TTL and cache warmer interval) by @mgdigital in https://github.com/bitmagnet-io/bitmagnet/pull/99
* Add reclassify command by @mgdigital in https://github.com/bitmagnet-io/bitmagnet/pull/100
* Disable Gorm easer plugin which sometimes causes zero results by @mgdigital in https://github.com/bitmagnet-io/bitmagnet/pull/101
* Bump follow-redirects from 1.15.2 to 1.15.4 in /webui by @dependabot in https://github.com/bitmagnet-io/bitmagnet/pull/102
* Fix search ordering by @mgdigital in https://github.com/bitmagnet-io/bitmagnet/pull/104
* Refactor torrent processing by @mgdigital in https://github.com/bitmagnet-io/bitmagnet/pull/106
* Add 0.5.0 upgrade hook to queue reprocessing of all torrents by @mgdigital in https://github.com/bitmagnet-io/bitmagnet/pull/107

**Full Changelog**: https://github.com/bitmagnet-io/bitmagnet/compare/v0.4.1...v0.5.0-beta.1

<hr /><em>This discussion was created from the release <a href='https://github.com/bitmagnet-io/bitmagnet/releases/tag/v0.5.0-beta.1'>v0.5.0-beta.1</a>.</em>

**Comment by johnyb0y** [comment]
Thank you!
